iam-login-service/src/test/java/it/infn/mw/iam/test/oauth/scope/ScopeRegistryTests.java

```diff
@@ -43,46 +46,59 @@
 @RunWith(MockitoJUnitRunner.class)
 public class ScopeRegistryTests {
 
-  
+
   @Mock
   ClientDetails client;
-  
+
+  @Mock
+  SystemScopeRepository scopeRepo;
+
+  @Before
+  public void setup() {
+    SystemScope testScope = new SystemScope("test:/whatever");
+    when(scopeRepo.getAll()).thenReturn(Sets.newHashSet(testScope));
+  }
+
   @Test
   public void testEmptyScopes() {
-    
-    DefaultScopeMatcherRegistry matcherRegistry = new DefaultScopeMatcherRegistry(emptySet());
-    
-    when(client.getScope()).thenReturn(Sets.newHashSet("openid","profile"));
+
+    DefaultScopeMatcherRegistry matcherRegistry =
+        new DefaultScopeMatcherRegistry(emptySet(), scopeRepo);
+
+    when(client.getScope()).thenReturn(Sets.newHashSet("openid", "profile"));
     Set<ScopeMatcher> matchers = matcherRegistry.findMatchersForClient(client);
-    
+
     assertThat(matchers, not(nullValue()));
     assertThat(matchers, hasSize(2));
     assertThat(matchers, hasItem(stringEqualsMatcher("openid")));
     assertThat(matchers, hasItem(stringEqualsMatcher("profile")));
   }
-  
+
   @Test
   public void testNonMatchingScope() {
-    
-    DefaultScopeMatcherRegistry matcherRegistry = new DefaultScopeMatcherRegistry(newHashSet(regexpMatcher("^test:/.*$")));
-    
-    when(client.getScope()).thenReturn(Sets.newHashSet("openid","profile"));
+
+    DefaultScopeMatcherRegistry matcherRegistry =
+        new DefaultScopeMatcherRegistry(newHashSet(regexpMatcher("^test:/.*$")), scopeRepo);
+
+    when(client.getScope()).thenReturn(Sets.newHashSet("openid", "profile"));
     Set<ScopeMatcher> matchers = matcherRegistry.findMatchersForClient(client);
-    
+
     assertThat(matchers, not(nullValue()));
     assertThat(matchers, hasSize(2));
     assertThat(matchers, hasItem(stringEqualsMatcher("openid")));
     assertThat(matchers, hasItem(stringEqualsMatcher("profile")));
   }
-  
+
   @Test
   public void testMatchingScope() {
-    
-    DefaultScopeMatcherRegistry matcherRegistry = new DefaultScopeMatcherRegistry(newHashSet(regexpMatcher("^test:/.*$")));
-    
-    when(client.getScope()).thenReturn(Sets.newHashSet("openid","profile", "test", "test:/whatever"));
+
+    DefaultScopeMatcherRegistry matcherRegistry =
+        new DefaultScopeMatcherRegistry(newHashSet(regexpMatcher("^test:/.*$")), scopeRepo);
+
+    when(client.getScope())
+      .thenReturn(Sets.newHashSet("openid", "profile", "test", "test:/whatever"));
     Set<ScopeMatcher> matchers = matcherRegistry.findMatchersForClient(client);
-    
+
```

Comment:
Try also a custom scope with a parametric path

Reply:
Another test could be when we have a scope policy on a parametric scope, and we define the same "parametric" custom scope.